dlib/crc32/crc32_kernel_1.h

```diff
@@ -26,10 +26,18 @@ namespace dlib
         // this is here for backwards compatibility with older versions of dlib.
         typedef crc32 kernel_1a;
 
-        inline crc32 (        
+        inline crc32 (
         );
 
-        inline crc32 (        
+        crc32 (
+            const dlib::crc32&
+        ) = default;
+
+        crc32& operator=(
+            const crc32&
+        ) = default;
```

Comment:
@davisking With warnings treated as errors, we got multiple issues along these lines:

`error: implicitly-declared ‘constexpr dlib::crc32::crc32(const dlib::crc32&)’ is deprecated [-Werror=deprecated-copy]`

In my understanding, this is because the "rule of three" is violated.

In some cases it is the assignment operator that is missing, in others it's the copy constructor.

I first added the missing functions, and when things worked, moved on to use compiler-generated versions instead (at least in most cases). Hope that's fine with you.

Reply:
Yeah not violating the rule of three is fine.  Although this class should just not declare any copy or assignment operators.

Reply:
Oh, ok. I can change that in the places I'm touching.

Reply:
In that case, remove them from `*_abstract.h`, too?

Reply:
Yep